dev/tools/gen_keycodes/lib/physical_key_data.dart

```diff
@@ -0,0 +1,382 @@
+      }
+      return MapEntry<String, List<int>>(name, keyCodes.toList()..sort());
+    });
+
+    return result;
+  }
+
+  /// Parses entries from Chromium's HID code mapping header file.
+  ///
+  /// Lines in this file look like this (without the ///):
+  ///            USB       evdev   XKB     Win     Mac     Code     Enum
+  /// DOM_CODE(0x000010, 0x0000, 0x0000, 0x0000, 0xffff, "Hyper", HYPER),
+  static Map<String, PhysicalKeyEntry> _readHidEntries(
+    String input,
+    Map<String, List<int>> nameToAndroidScanCodes,
+    Map<String, List<int>> nameToGlfwKeyCodes,
+  ) {
+    final Map<int, PhysicalKeyEntry> entries = <int, PhysicalKeyEntry>{};
+    final RegExp usbMapRegExp = RegExp(
+      r'DOM_CODE\s*\(\s*'
+      r'0x(?<usb>[a-fA-F0-9]+),\s*'
+      r'0x(?<evdev>[a-fA-F0-9]+),\s*'
+      r'0x(?<xkb>[a-fA-F0-9]+),\s*'
+      r'0x(?<win>[a-fA-F0-9]+),\s*'
+      r'0x(?<mac>[a-fA-F0-9]+),\s*'
+      r'(?:"(?<code>[^\s]+)")?[^")]*?,'
+      r'\s*(?<enum>[^\s]+?)\s*'
+      r'\)',
+      // Multiline is necessary because some definition spreads across
+      // multiple lines.
+      multiLine: true,
+    );
+    final RegExp commentRegExp = RegExp(r'//.*$', multiLine: true);
+    input = input.replaceAll(commentRegExp, '');
+    for (final RegExpMatch match in usbMapRegExp.allMatches(input)) {
+      final int usbHidCode = getHex(match.namedGroup('usb')!);
+      final int linuxScanCode = getHex(match.namedGroup('evdev')!);
+      final int xKbScanCode = getHex(match.namedGroup('xkb')!);
+      final int windowsScanCode = getHex(match.namedGroup('win')!);
+      final int macScanCode = getHex(match.namedGroup('mac')!);
+      final String? chromiumCode = match.namedGroup('code');
+      // The input data has a typo...
+      final String enumName = match.namedGroup('enum')!.replaceAll('MINIMIUM', 'MINIMUM');
+
+      final String name = chromiumCode ?? shoutingToUpperCamel(enumName);
+      if (name == 'IntlHash') {
+        // Skip key that is not actually generated by any keyboard.
+        continue;
+      }
+      final PhysicalKeyEntry newEntry = PhysicalKeyEntry(
+        usbHidCode: usbHidCode,
+        androidScanCodes: nameToAndroidScanCodes[name] ?? <int>[],
+        glfwKeyCodes: nameToGlfwKeyCodes[name] ?? <int>[],
+        linuxScanCode: linuxScanCode == 0 ? null : linuxScanCode,
+        xKbScanCode: xKbScanCode == 0 ? null : xKbScanCode,
+        windowsScanCode: windowsScanCode == 0 ? null : windowsScanCode,
+        macOsScanCode: macScanCode == 0xffff ? null : macScanCode,
+        iosScanCode: (usbHidCode & 0x070000) == 0x070000 ? (usbHidCode ^ 0x070000) : null,
+        name: name,
+        chromiumCode: chromiumCode,
+      );
+      // Remove duplicates: last one wins, so that supplemental codes
+      // override.
+      if (entries.containsKey(newEntry.usbHidCode)) {
+        // This is expected for Fn. Warn for other keys.
+        if (!<String>{'Fn'}.contains(newEntry.name)) {
```

Comment:
Why can't this just be written this way?:
```suggestion
        if (newEntry.name != 'Fn') {
```